src/text_shaper/font_locator_provider.cpp

```diff
@@ -18,32 +18,20 @@ font_locator_provider& font_locator_provider::get()
     return instance;
 }
 
-#if defined(__APPLE__)
-font_locator& font_locator_provider::coretext()
-{
-    if (!_coretext)
-        _coretext = make_unique<coretext_locator>();
-
-    return *_coretext;
-}
-#endif
-
-#if defined(_WIN32)
-font_locator& font_locator_provider::directwrite()
+font_locator& font_locator_provider::native()
 {
-    if (!_directwrite)
-        _directwrite = make_unique<directwrite_locator>();
-
-    return *_directwrite;
-}
+    if (!_native)
+    {
+#if defined(__APPLE__)
+        _native = make_unique<coretext_locator>();
+#elif defined(_WIN32)
+        _native = make_unique<directwrite_locator>();
+#else
+        _native = make_unique<fontconfig_locator>();
```

Comment:
Can we move it into  constructor?

Reply:
Hmm.... on CI we don't call `native()` but `mock()` and thus, this would usually never get instanciated. Moving it to the constructor, would do so, howeve.r Not sure we want that (I think, as of today, it would not hurt though)